### PR TITLE
Marvell prestera kernel driver

### PR DESCRIPTION
**Pull request by antony-rheneus**
Build Marvell kernel driver for prestera sai sdk
Builds interrupt and dma kernel driver as part of sonic build

mvIntDrv.ko and mvDmaDrv.ko modules are loaded using etc/modules-load.d/marvell.conf script 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Comment by lguohan** [comment]
can you remove replaced kernel driver?

File: platform/marvell-armhf/prestera/debian/rules [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+clean:
+	dh_testdir
+	dh_testroot
+	dh_clean
+
+build:
+	# get sources
+	rm -rf mrvl-prestera || true
+	git clone https://github.com/Marvell-switching/mrvl-prestera.git
```
Comment (lguohan) [inline platform/marvell-armhf/prestera/debian/rules:22]: we need tag to checkout a specific commit.

otherwise, the build is not reproducible.

**Comment by antony-rheneus** [comment]
> can you remove replaced kernel driver binary?

Ok sure, I will remove it, and add as part of this PR too.

File: platform/marvell-armhf/prestera/debian/rules (hunk 1 above)
Comment (antony-rheneus) [reply platform/marvell-armhf/prestera/debian/rules:22]: Ok sure, I will add git tag, and update here accordingly

Comment (antony-rheneus) [reply platform/marvell-armhf/prestera/debian/rules]: Addressed your comment, by adding git tag, and updated the git clone url with the TAG.

**Comment by antony-rheneus** [comment]
> > can you remove replaced kernel driver binary?
> 
> Ok sure, I will remove it, and add as part of this PR too.

Done, removed the kernel binary  package by removing all its makefile content
Now it is replaced with the prestera.mk which will download the source and build inside sonic buildimage.

**Comment by antony-rheneus** [comment]
retest this please

**Comment by antony-rheneus** [comment]
retest this please

**Comment by antony-rheneus** [comment]
@lguohan , I hav addressed your comments, on maintaining git tag for the driver source, as well removing old ko binaries from the build.
Would you please review and confirm?

File: platform/marvell-armhf/rules.mk [hunk 2]
```diff
@@ -5,9 +5,9 @@ include $(PLATFORM_PATH)/docker-syncd-mrvl-rpc.mk
 include $(PLATFORM_PATH)/docker-saiserver-mrvl.mk
 include $(PLATFORM_PATH)/libsaithrift-dev.mk
 include $(PLATFORM_PATH)/one-image.mk
-include $(PLATFORM_PATH)/linux-kernel-armhf.mk
```
Comment (lguohan) [inline platform/marvell-armhf/rules.mk:8]: can you remove this file if it is not used.

Comment (antony-rheneus) [reply platform/marvell-armhf/rules.mk:8]: Thanks for pointing it out, I have removed the obsolete files now.

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
can we get this one to 202012 branch?

**Comment by antony-rheneus** [comment]
> can we get this one to 202012 branch?

Ok sure, I will port it to 202012 branch.

**Pull request merged**